feat: switch to npm as default package manager

deepak1556 opened the pull request:
Fixes https://github.com/microsoft/vscode/issues/196795

deepak1556 committed:
feat: move from yarn to npm

deepak1556 committed:
chore: skip yarn.lock files

deepak1556 committed:
fix: playwright download

deepak1556 committed:
chore: fix compile and hygiene

deepak1556 committed:
chore: bump vsce@2.17.0

Refs https://github.com/microsoft/vscode-vsce/commit/8b49e9dfdf909ad3af2b9ec9c825f5b501f6d75e

deepak1556 committed:
test: update results for bat and sh colorizer tests

deepak1556 committed:
fix: add missing lock files for windows

deepak1556 committed:
fix: switch to legacy-peer-deps

deepak1556 committed:
chore: update markdown-it@14.1.0

Refs https://github.com/markdown-it/markdown-it/commit/737c95a12976357df99652e4b51d831cac4a75aa

esbuild step in extensions-ci-pr was previously using markdown-it
from root which had userland punycode and was able to compile successfully.

deepak1556 committed:
ci: increase pr timeout for windows integration tests

deepak1556 committed:
chore: fix product build

deepak1556 committed:
build: ignore extension dev dependency for rcedit

deepak1556 committed:
build: fix working directory inside container

deepak1556 committed:
build: fix dependency generation

deepak1556 committed:
npm: update dependencies

deepak1556 committed:
ci: use global npmrc

deepak1556 committed:
ci: update cache

deepak1556 committed:
ci: setup global npmrc for private npm auth

deepak1556 committed:
build: fix extension bundling

deepak1556 committed:
chore: sync npm dependencies

deepak1556 committed:
ci: debug env variables for container

deepak1556 committed:
ci: fix win32 cli pipeline

deepak1556 committed:
build: fix npmrc config usage for build/ and remote/ dirs

deepak1556 committed:
fix: windows build

deepak1556 committed:
fix: container builds

deepak1556 committed:
fix: markdown-language-features tests and bundling

```
[03:58:22] Error: Command failed: /Users/demohan/.nvm/versions/node/v20.15.1/bin/node /Users/demohan/github/vscode/extensions/markdown-language-features/esbuild-notebook.js --outputRoot /Users/demohan/github/vscode/.build/extensions/markdown-language-features
✘ [ERROR] Could not resolve "punycode"

    extensions/markdown-language-features/node_modules/markdown-it/lib/index.js:14:27:
      14 │ var punycode     = require('punycode');
         ╵                            ~~~~~~~~~~

  The package "punycode" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.
```

Adds userland package based on https://github.com/markdown-it/markdown-it/commit/beed9aee2c1b44819afc25d485e86a6c699b6ef0

deepak1556 committed:
fix: container builds for distro

deepak1556 committed:
chore: update yarn occurrences

deepak1556 committed:
fixup! chore: bump vsce@2.17.0

Uses the closest version to `main` branch that does not
include https://github.com/microsoft/vscode-vsce/commit/d3cc84cdec99b8e9f24be63d1a2a54abe908f68b
while still having the fix https://github.com/microsoft/vscode-vsce/commit/8b49e9dfdf909ad3af2b9ec9c825f5b501f6d75e

deepak1556 committed:
chore: sync npm dependencies

deepak1556 committed:
chore: sync npm dependencies

deepak1556 committed:
chore: sync npm dependencies

deepak1556 committed:
chore: throw error when yarn is used for installation

joaomoreno commented on extensions/css-language-features/CONTRIBUTING.md:
```diff
@@ -2,13 +2,13 @@
 
 - Clone [microsoft/vscode](https://github.com/microsoft/vscode)
-- Run `yarn` at `/`, this will install
+- Run `npm ci` at `/`, this will install
```
```suggestion
- Run `npm install` at `/`, this will install
```

joaomoreno commented on extensions/css-language-features/CONTRIBUTING.md:
```diff
@@ -30,14 +30,14 @@ However, within this extension, you can run a development version of `vscode-css
-- Run `yarn` in `vscode-css-languageservice`
-- Run `yarn link` in `vscode-css-languageservice`. This will compile and link `vscode-css-languageservice`
-- In `css-language-features/server/`, run `yarn link vscode-css-languageservice`
+- Run `npm ci` in `vscode-css-languageservice`
```
```suggestion
- Run `npm install` in `vscode-css-languageservice`
```

joaomoreno commented on extensions/html-language-features/.vscode/tasks.json:
```diff
@@ -2,8 +2,8 @@
-			"label": "yarn",
-			"command": "yarn",
+			"label": "npm install",
+			"command": "npm ci",
```
```suggestion
			"command": "npm install",
```

joaomoreno commented on extensions/html-language-features/CONTRIBUTING.md:
```diff
@@ -1,12 +1,12 @@
 
 - Clone [microsoft/vscode](https://github.com/microsoft/vscode)
-- Run `yarn` at `/`, this will install
+- Run `npm ci` at `/`, this will install
```
```suggestion
- Run `npm install` at `/`, this will install
```

joaomoreno commented on extensions/html-language-features/CONTRIBUTING.md:
```diff
@@ -28,13 +28,13 @@ However, within this extension, you can run a development version of `vscode-htm
-- Run `yarn` in `vscode-html-languageservice`
-- Run `yarn link` in `vscode-html-languageservice`. This will compile and link `vscode-html-languageservice`
-- In `html-language-features/server/`, run `yarn link vscode-html-languageservice`
+- Run `npm ci` in `vscode-html-languageservice`
```
```suggestion
- Run `npm install` in `vscode-html-languageservice`
```

joaomoreno commented on extensions/json-language-features/CONTRIBUTING.md:
```diff
@@ -29,13 +29,13 @@ However, within this extension, you can run a development version of `vscode-jso
 
 - Clone [microsoft/vscode-json-languageservice](https://github.com/microsoft/vscode-json-languageservice)
-- Run `npm install` in `vscode-json-languageservice`
+- Run `npm ci` in `vscode-json-languageservice`
```
```suggestion
- Run `npm install` in `vscode-json-languageservice`
```

joaomoreno commented on extensions/markdown-language-features/package.json:
```diff
@@ -770,6 +770,8 @@
     "markdown-it-front-matter": "^0.2.4",
     "morphdom": "^2.6.1",
     "picomatch": "^2.3.1",
+    "punycode": "^2.3.1",
+    "tslib": "*",
     "vscode-languageclient": "^8.0.2",
```
What happened here?

deepak1556 replied:
The `tslib` is a leftover to satisfy peer-deps which we can remove with the `legacy-peer-deps` flags enabled.

As for `punycode`, it is to resolve the following error https://github.com/microsoft/vscode/pull/226927/commits/3b2cd961553df4e87172ca17b764c2741652b437. We can remove it once we update to `markdown-it >= 14.1.0`

joaomoreno commented on extensions/typescript-language-features/web/README.md:
```diff
@@ -6,8 +6,8 @@ Language server host for typescript using vscode's sync-api in the browser.
 
-1. `yarn watch` for vscode itself
-2. `yarn watch-web` for the web side
+1. `npm ci` for vscode itself
```
```suggestion
1. `npm install` for vscode itself
```

joaomoreno commented on extensions/vscode-colorize-tests/test/colorize-fixtures/test.bat:
```diff
@@ -6,7 +6,7 @@ title VSCode Dev
 
 :: Node modules
-if not exist node_modules call .\scripts\npm.bat install
+if not exist node_modules call npm ci
```
```suggestion
if not exist node_modules call npm install
```

joaomoreno commented on extensions/vscode-colorize-tests/test/colorize-fixtures/test.sh:
```diff
@@ -19,7 +19,7 @@ function code() {
 
 	# Node modules
-	test -d node_modules || ./scripts/npm.sh install
+	test -d node_modules || npm ci
```
```suggestion
	test -d node_modules || npm install
```

joaomoreno commented on test/smoke/README.md:
```diff
@@ -34,8 +34,9 @@ You must always run the smoketest version that matches the release you are testi
 git checkout release/1.22
-yarn && yarn compile
-yarn --cwd test/smoke
+npm ci && npm run compile
```
```suggestion
npm install && npm run compile
```

joaomoreno reviewed (commented):
Good to merge once ready for me, once suggested changes are adopted. Also, let's go ahead and delete the yarn.lock files in the same PR.

I will be out so please someone else give Deepak the official green checkbox when it's time to merge.

deepak1556 committed:
chore: add review feedback

deepak1556 committed:
chore: switch exec => run where needed

deepak1556 committed:
chore: npm sync dependencies

deepak1556 commented on build/azure-pipelines/darwin/product-build-darwin-test.yml:
```diff
@@ -12,7 +12,7 @@ parameters:
 
 steps:
-  - script: yarn npm-run-all -lp "electron $(VSCODE_ARCH)" "playwright-install"
+  - script: npm exec -- npm-run-all -lp "electron $(VSCODE_ARCH)" "playwright-install"
```
For commands from other packages `exec` is preferred. Commands from `scripts` config of `package.json` can use `run`.

deepak1556 committed:
fix: markdown-language-features bundling

```
✘ [ERROR] Could not resolve "punycode"

    extensions/markdown-language-features/node_modules/markdown-it/lib/index.js:14:27:
      14 │ var punycode     = require('punycode');
         ╵                            ~~~~~~~~~~

  The package "punycode" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.
```

Adds missing userland package based on markdown-it/markdown-it@beed9ae,
can be removed once we update markdown-it >= 14.1.0

deepak1556 committed:
ci: rename no-yarn-lock-changes.yml

deepak1556 committed:
chore: sync npm dependencies

deepak1556 committed:
ci: restore no-yarn-lock-changes.yml

We can disable it in a separate PR to keep the required
checks happy and also need workflow edit perms.

deepak1556 committed:
chore: sync npm dependencies

deepak1556 committed:
ci: rebuild cache

deepak1556 committed:
ci: fix no-package-lock-changes.yml

deepak1556 committed:
chore: bump distro

deepak1556 committed:
chore: rm yarn.lock files

deepak1556 committed:
chore: rm yarn.lock files without dependencies

deepak1556 committed:
chore: add vscode-selfhost-import-aid to postinstall dirs

deepak1556 committed:
chore: bump distro

jwoo0122 commented:
We should change [the contribution guide](https://github.com/microsoft/vscode/wiki/How-to-Contribute#build-and-run). It still directs contributors to use yarn.

brjsp commented:
Speaking as a downstream redistributor, that's precisely the kind of change that warrants bumping up the major version of your software.

deepak1556 commented:
I can't speak for the versioning, you should still be able to use the yarn version of commands to launch any of the scripts that were previously used for building packages. The only command that needs a change is the dependency installation `yarn -> npm i` if you want to keep the changes minimal following this PR. Let me know if you are facing any other issues due to the transition and I can help were possible.

liamlows commented:
Is there a planned tag release for this change? We are also using vscode-web and would like to use npm when building but the latest tag (1.93.1) seems to still use yarn.

deepak1556 commented:
@liamlows it will be available from `1.94.0`
